dev/conductor/ui/lib/main.dart

```diff
@@ -2,24 +2,35 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io' as io;
 import 'package:conductor_core/conductor_core.dart';
 import 'package:conductor_core/proto.dart' as pb;
-
+import 'package:desktop_window/desktop_window.dart';
 import 'package:file/file.dart';
 import 'package:file/local.dart';
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:platform/platform.dart';
 
-const String _title = 'Flutter Conductor';
+import 'widgets/progression.dart';
+
+const String _title = 'Flutter Desktop Conductor (Not ready, do not use)';
 
 const LocalFileSystem _fs = LocalFileSystem();
 const LocalPlatform _platform = LocalPlatform();
 final String _stateFilePath = defaultStateFilePath(_platform);
 
-void main() {
+Future<void> main() async {
   final File _stateFile = _fs.file(_stateFilePath);
-  final pb.ConductorState? state = _stateFile.existsSync() ? readStateFromFile(_stateFile) : null;
-  runApp(MyApp(state));
+  final pb.ConductorState? state =
+      _stateFile.existsSync() ? readStateFromFile(_stateFile) : null;
+
+  WidgetsFlutterBinding.ensureInitialized();
+  // app currently only supports macOS and Linux, and a minimum app size is added to prevent overflow
+  if (!kIsWeb && (io.Platform.isMacOS || io.Platform.isLinux)) {
```

Comment:
we should handle if this is false. I would actually check if it's *not* a supported platform, and then throw with a helpful message in that situation.

Reply:
Ok, you mean another unit test to check if the platform is correct?

Reply:
No, I mean right here. If we ever reach this point on web, the app will just close, and the user won't know why.

Reply:
You could do something like this at the start of main to exit
```
if (!(io.Platform.isMacOS || io.Platform.isLinux)) {
  throw Exception('The release dashboard is only support on MacOS and Linux Desktop');
}
```

You can rely on the conductor package failing when they try to build web (dart:io won't be available and will be a compilation error)

Reply:
I made an error widget for this.